includes/course-theme/class-sensei-course-theme-quiz.php

```diff
@@ -142,4 +143,40 @@ private static function render_contact_teacher() {
 		return $block->render_contact_teacher_block( null, $link );
 	}
 
+	/**
+	 * Maybe add quiz prerequisites notice.
+	 */
+	private function maybe_add_quiz_prerequisite_notice() {
```

Comment:
Most of the code here is the same with `Sensei_Course_Theme_Lesson::maybe_add_lesson_prerequisite_notice`. Does it make sense to move it in a commone, e.g. `Lesson_Prerequisite_Notice`?

Reply:
I thought about refactoring this a bit, but then there are also `Sensei_Course::get_course_prerequisite_message` and `Sensei_Lesson::prerequisite_complete_message` with similar logic in them. And those are only the parts that output the prerequisite messages. We have a lot of logic scattered around the Sensei codebase related to course/lesson/quiz prerequisites. If we decide to refactor then we should do it in those places too, otherwise I don't see much benefit.

But you're also right about not duplicating the same method with small changes in it. For this particular issue, I think we could simply reuse the message that we show in the lesson page. The issue is an edge case anyway. Because we don't actually show the quiz link unless the prerequisites of the lesson is not complete. So students almost never should visit a quiz page of a lesson that has incomplete prerequisites.

Updated to simply show the lesson prerequisite notice on the quiz page.

Reply:
Here is how it looks like now. Let me know what you think @gkaragia .

Prerequisite message for the lesson.
<img width="1709" alt="Screen Shot 2022-08-29 at 12 38 19" src="https://user-images.githubusercontent.com/2578542/187160430-40bba1f1-1a55-4743-a94d-a550a1405a29.png">

Prerequisite message for the quiz of the above lesson.
<img width="1710" alt="Screen Shot 2022-08-29 at 12 38 05" src="https://user-images.githubusercontent.com/2578542/187160414-1e33625b-b212-419c-8867-0f51defb3713.png">

Reply:
Updated the messaging to apply to both lessons and quizzes. 363e8c7b02f69e1e776f13b9a787baacfc60669f

Updated the `sensei_user_quiz_status_message` method to include quiz prerequisite notice message. 135c4cf19da0c3f2742dd63582b50d83bf43bedd

Reply:
> Hey @dadish my phrasing was not correct. What I meant to say with
> 
> > I think it's totally fine to change the wording a bit
> 
> was that it's ok to have the lesson wording in quizzes. I was comparing it to the previous version of your PR where you had a quiz-specific wording in the notice. Is it possible to revert [363e8c7](https://github.com/Automattic/sensei/commit/363e8c7b02f69e1e776f13b9a787baacfc60669f)? Sorry for the confusion.

Ah, makes sense. Reverted. 👍